backend/db_migrations.py

```
@@ -342,7 +404,100 @@ def apply_migration(conn: sqlite3.Connection, from_version: int, to_version: int
+                    "CREATE UNIQUE INDEX IF NOT EXISTS idx_cancer_subtype_unique ON cancer_subtypes(cancer_type_id, name)",
+                    "CREATE INDEX IF NOT EXISTS idx_cancer_subtype_drugs_subtype ON cancer_subtype_drugs(subtype_id)",
+                    "CREATE INDEX IF NOT EXISTS idx_cancer_subtype_drugs_chembl ON cancer_subtype_drugs(chembl_id)",
+                    "CREATE INDEX IF NOT EXISTS idx_cancer_subtype_drugs_score ON cancer_subtype_drugs(subtype_id, rank_score)",
+                    "CREATE UNIQUE INDEX IF NOT EXISTS idx_cancer_subtype_drugs_unique ON cancer_subtype_drugs(subtype_id, chembl_id)",
+                ]
+                for ddl in migration_7_ddl:
+                    try:
+                        cursor.execute(ddl)
+                    except sqlite3.OperationalError as e:
+                        if 'already exists' in str(e).lower():
+                            print(f"   ⚠️  Table already exists, skipping")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)